posthog/api/file_system.py

```diff
@@ -186,7 +186,15 @@ def _apply_search_to_queryset(self, queryset: QuerySet, search: str) -> QuerySet
                     q = Q(path__icontains=token)
             else:
                 # plain free-text token: search in path
-                q = Q(path__icontains=token)
+                # If the token contains a slash, treat it as a path and search for each segment
+                if "/" in token:
+                    path_parts = token.split("/")
+                    q = Q()
+                    for part in path_parts:
+                        if part:  # Skip empty parts
+                            q &= Q(path__icontains=part)
+                else:
+                    q = Q(path__icontains=token)
```

Comment:
Backend changes require backend tests ;).

That said... perhaps an easier solution is to also "escape" the entire search string? So that `go/reventue` searches for `go\/revenue` and finds it? Though I can also see people wanting to search "unfiled/product/pineapple" to search for parts within folders... so this works for me too.